pkg/probe/exec/exec.go

```diff
@@ -39,7 +44,34 @@ type execProber struct{}
 // from executing a command. Returns the Result status, command output, and
 // errors if any.
 func (pr execProber) Probe(e exec.Cmd) (probe.Result, string, error) {
-	data, err := e.CombinedOutput()
+	stdoutPipe, err := e.StdoutPipe()
+	if err != nil {
+		return probe.Failure, "", err
+	}
+	stderrPipe, err := e.StderrPipe()
+	if err != nil {
+		return probe.Failure, "", err
+	}
+
+	stdout := make(chan string)
+	stderr := make(chan string)
+
+	go func() {
+		b, _ := utilio.ReadAtMost(stdoutPipe, maxReadLength)
+		stdout <- string(b)
+	}()
+	go func() {
+		b, _ := utilio.ReadAtMost(stderrPipe, maxReadLength)
+		stderr <- string(b)
+	}()
+
+	if err := e.Start(); err != nil {
+		return probe.Failure, "", err
+	}
+
+	err = e.Wait()
```

Comment:
will this block until output is consumed? Do we need to drain stdout/stderr?

Reply:
good point, https://golang.org/pkg/os/exec/#Cmd.Wait says

```
If any of c.Stdin, c.Stdout or c.Stderr are not an *os.File, Wait also waits for the respective I/O loop copying to or from the process to complete.
```

Reply:
gotta test more tomorrow

Reply:
updated code and tested as well